src/LendingPool.sol

```diff
@@ -1247,6 +1249,13 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, ILendingPool {
         }
     }
 
+    /**
+     * @inheritdoc Creditor
+     */
+    function closeMarginAccount(address account) external view override {
+        if (maxWithdraw(account) != 0) revert OpenPositionNonZero();
```

Comment:
if this will check on maxWithdraw, the getopenposition in the account should be removed

Reply:
getOpenPosition is used in different places to for example in getUsedMargin in accounts. I would do as a totally seperate task to refactor.